scipy-1.0/ckdtree.tex

```diff
@@ -0,0 +1,35 @@
+The cKDTree module is rewritten in C++ with templated classes.
+
+The support to a box of periodic boundary condition is added. A periodic
+condition is typically used in computer simulations of physical processes.
+
+\subsubsection{k-NN}
+
+The API for querying neighbours is enhanced, to allow querying a subset of
+k nearest neighbours.
+
+\subsubsection{sparse distance matrix}
+
+generating approximated sparse distance matrix
+
+\subsubsection{Pair-Counting}
+
+The cKDTree module implements the dual tree counting algorithm in (cite).
```

Comment:
The docstring cites `Maneewongvatana and Mount 1999`, which is perhaps: https://arxiv.org/abs/cs/9901013

Conversely, the only reference I see to `dual-tree` in the source code is for `Gray and Moore` : https://arxiv.org/abs/astro-ph/0012333